src/framework/application.js

```diff
@@ -2318,6 +2313,20 @@ var makeTick = function (_app) {
         application.fire("framerender");
 
         if (application.autoRender || application.renderNextFrame) {
+            // In AUTO mode the resolution is changed to match the canvas size
+            if (application._resolutionMode === RESOLUTION_AUTO) {
+                // Check if the canvas DOM has changed size
+                const canvas = application.graphicsDevice.canvas;
+                const pixelRatio = application.graphicsDevice.maxPixelRatio;
+
+                const clientWidth = canvas.clientWidth;
+                const clientHeight = canvas.clientHeight;
+
+                if (canvas.width !== (clientWidth * pixelRatio) || canvas.height !== (clientHeight * pixelRatio)) {
```

Comment:
I wonder if some rounding should be used here? I assume the pixelRatio is float and so when you multiply clientWidth by it, you need to round it to whole number? Strange that even resizeCanvas function does not do that. can canvas.width be set to fractional number?

Reply:
Code from WebGL fundamentals have the following:
```
  /**
   * Resize a canvas to match the size its displayed.
   * @param {HTMLCanvasElement} canvas The canvas to resize.
   * @param {number} [multiplier] amount to multiply by.
   *    Pass in window.devicePixelRatio for native pixels.
   * @return {boolean} true if the canvas was resized.
   * @memberOf module:webgl-utils
   */
  function resizeCanvasToDisplaySize(canvas, multiplier) {
    multiplier = multiplier || 1;
    const width  = canvas.clientWidth  * multiplier | 0;
    const height = canvas.clientHeight * multiplier | 0;
    if (canvas.width !== width ||  canvas.height !== height) {
      canvas.width  = width;
      canvas.height = height;
      return true;
    }
    return false;
  }
```

I can't find anything yet about whether width/height can be non-integers

Reply:
Interesting that:
```
canvas.clientWidth  * multiplier | 0
```
The `| 0` rounds to a whole number

Reply:
I will need to do more investigation as there looks like there can be edge cases to deal with. This will not go out with 1.41.0 and moving to draft while I figure out what is the best route here 😅

Reply:
clientWidth and clientHeight should return int, but getBoundingClientRect - returns object with floats. And this should be taken into an account in such context.

In this case, you definitely want to `floor` this: `clientWidth * pixelRatio`, and then compare to `canvas.width`, as setting float to `canvas.width`, it will `floor` it for you.

Reply:
I've explicitly floored both the check and the assignment on the canvas (just in case some browsers round or something). :)